gitlab-runner-mock/src/lib.rs

```diff
@@ -171,4 +181,23 @@ impl GitlabRunnerMock {
     pub fn set_update_interval(&self, interval: u32) {
         *self.inner.update_interval.lock().unwrap() = interval;
     }
+
+    pub(crate) fn expected_metadata(&self) -> ExpectedMetadata {
+        self.inner.expected_metadata.lock().unwrap().clone()
+    }
+
+    pub fn set_expected_metadata(
+        &self,
+        version: Option<String>,
```

Comment:
Could this just take an `ExpectedMetadata` argument? Passing four args of the exact same type seems like it would be easy to mix them all up.

Reply:
moved to a buildery api to set one at a time